src/rmarkdown/preview.ts

```diff
@@ -24,12 +24,12 @@ class RMarkdownPreview extends vscode.Disposable {
 
     constructor(title: string, cp: cp.ChildProcessWithoutNullStreams, panel: vscode.WebviewPanel,
         resourceViewColumn: vscode.ViewColumn, outputUri: vscode.Uri, uri: vscode.Uri,
-        RMarkdownPreviewManager: RMarkdownPreviewManager, themeBool: boolean, autoRefresh: boolean) {
+        RMarkdownPreviewManager: RMarkdownPreviewManager, useDarkTheme: boolean, autoRefresh: boolean) {
         super(() => {
             this.cp?.kill('SIGKILL');
             this.panel?.dispose();
             this.fileWatcher?.close();
-            fs.removeSync(this.outputUri.path);
+            fs.removeSync(this.outputUri.fsPath);
```

Comment:
Would you like to try working with an untitled document with rmd language selected? How should we handle this case? Let user save the document to get a valid `fsPath` first, or we write the document to a temp file and render that file as if we render this untitled document?

Reply:
It might be a bit tricky because we would have to watch for changes on the untitled rmd file in order to sync it with the temp rmd file. I'm not aware of methods for tracking changes on virtual docs

Reply:
The error is like the following with untitled rmd:

```
[VSC-R] Untitled-1 process started
Error in abs_path(input) : The file 'Untitled-1' does not exist.
Calls: cat -> <Anonymous> -> setwd -> dirname -> abs_path
In addition: Warning message:
In normalizePath(path, winslash = winslash, mustWork = mustWork) :
  path[1]="Untitled-1": No such file or directory

Backtrace:

1: stop("The file '", x, "' does not exist.")

2: abs_path(input)
3: dirname(abs_path(input))
4: setwd(dirname(abs_path(input)))

5: rmarkdown::render("Untitled-1", output_format = rmarkdown::html_document(), 

6: cat("---vsc---", rmarkdown::render("Untitled-1", output_format = rmarkdown::html_document(), 

Exiting on error

[VSC-R] Untitled-1 process exited with exit code 1
```

Reply:
Right, that would be relatively straight forward to solve. Given the auto refresh is based on changes to a saved file, having it auto refresh would be the difficult part. ...and I'm not sure it would make sense since the file wouldn't be untitled anymore once saved.

Reply:
Looks like we should ask the user to save the file first. If the document is untitled, then we don't render it.